lib/components/user/user-account-screen.js

```diff
@@ -0,0 +1,437 @@
+
+      if (!isNewAccount && skipIfExistingUser) {
+        // Return to originalUrl and prevent the URL from showing in browsing history
+        window.location.replace(window.location.href.replace('#/account/welcome', `#${originalUrl}`))
+      } else {
+        // TODO: Set application state (so that user data is fetched once only... until logout happens).
+        const userData = isNewAccount ? getStateForNewUser(auth.user) : result.data
+        if (userData.savedLocations === null) userData.savedLocations = []
+
+        this.setState({
+          isNewAccount,
+          userData
+        })
+      }
+    } catch (error) {
+      // TODO: improve error handling.
+      alert(`An error was encountered:\n${error}`)
+    }
+  }
+
+  _updateUserState = newUserData => {
+    const { userData } = this.state
+    this.setState({
+      userData: {
+        ...userData,
+        ...newUserData
+      }
+    })
+  }
+
+  _updateUserPrefs = async () => {
+    const { auth, persistence } = this.props
+
+    if (persistence && persistence.otp_middleware) {
+      const { accessToken } = auth
+      const { isNewAccount, userData } = this.state
+
+      // TODO: Change state of Save button.
+
+      let result
+      if (isNewAccount) {
+        result = await addUser(persistence.otp_middleware, accessToken, userData)
+      } else {
+        result = await updateUser(persistence.otp_middleware, accessToken, userData)
+      }
+
+      // TODO: improve this.
+      if (result.status === 'success') {
+        alert('Your preferences have been saved.')
+      } else {
+        alert(`An error was encountered:\n${result.message}`)
+      }
+    }
+  }
+
+  _handleConsentChange = e => {
+    this._updateUserState({ hasConsentedToTerms: e.target.checked })
+  }
+
+  _handleExit = () => {
+    const { originalUrl } = this.props
+    this.props.routeTo(originalUrl)
+  }
+
+  _handleExitAndSave = async () => {
+    await this._updateUserPrefs()
+    this._handleExit()
+  }
+
+  _handleLocationsChange = newLocations => {
+    this._updateUserState({ savedLocations: newLocations })
+  }
+
+  _handleNotificationChannelChange = e => {
+    this._updateUserState({ notificationChannel: e })
+  }
+
+  _handleToNextPane = panes => {
+    return async () => {
+      const { activePaneId } = this.state
+      const nextId = panes[activePaneId].nextId
+
+      if (nextId) {
+        this.setState({
+          activePaneId: nextId
+        })
+      } else {
+        await this._handleExitAndSave()
+      }
+    }
+  }
+
+  _handlePhoneNumberChange = e => {
+    this._updateUserState({ phoneNumber: e.target.value })
+  }
+
+  _handleStoreTripsChange = e => {
+    this._updateUserState({ storeTripHistory: e.target.checked })
+  }
+
+  _handleToPrevPane = panes => {
+    return () => {
+      const { activePaneId } = this.state
+      this.setState({
+        activePaneId: panes[activePaneId].prevId
+      })
+    }
+  }
+
+  _makePanes = () => {
```

Comment:
This generates way too much react that isn't needed on every render. It should be refactored to be something like: 

```js
renderActivePane = () => {
  switch(this.state.activePaneId) {
    case 'terms': return this.renderTerms()
    case 'notifications': return this.renderNotifications()
    ...
  }
}
```